src/App.tsx

```diff
@@ -82,8 +84,39 @@ function App() {
     resetButton()
   }
 
+  const csvData = ():(string | number)[][]  => { 
+    const data = hrsArr.map(f =>
+      [ f.date.toLocaleDateString('en-us', { timeZone: 'UTC' }),
+        f.hrs,
+        f.totalHrs
+      ])
+    
+    return [
+      ['Date', 'Hours', 'Running Total'],
+      ...data
+    ]
+  }
+
+  const handleDownload = async () => {
+    setDownload(() => true)
+    setTimeout(() => setDownload(() => false), 2000)
+  }
+
   const table = (
     <div className="table">
+      <div className="export-button">
+        {download ? 
+          <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
+            <CSVDownload data={csvData()} target="_self" />
```

Comment:
From the [react-csv docs](https://github.com/react-csv/react-csv?tab=readme-ov-file#--onclick-props) (see `this.dataFromListOfUsersState`, it looks like you might be able to pass the function itself to `data` instead of calling it in the render. This way, it only gets called on click.